src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        //eventBridgeStarterAppTemplate requires aws credentials
+        if (val && val.label === eventBridgeStarterAppTemplate) {
+            if (this.credentialsNotFound) {
+                await ext.awsContextCommands.onCommandLogin()
+
+                return exitWizard
+            }
+
+            //check if detected samCliVersion supports eventBridgeStarterAppTemplate
+            if (semver.lt(this.samCliVersion, MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE)) {
+                vscode.window.showErrorMessage(
+                    localize(
+                        'AWS.samcli.error.invalid_schema_support_version',
+                        'Installed SAM executable does not support templates that require Event Schema selection. Required minimum version {0}, but found {1}',
+                        MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE,
+                        this.samCliVersion
+                    )
+                )
+
+                return exitWizard
+            }
+        }
+
+        return val ? (val.label as SamTemplate) : undefined
+    }
+
+    public async promptUserForRegion(currRegion?: string): Promise<string | undefined> {
+        const schemasRegions = ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-northeast-1']
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.schemas.region.prompt', 'Select a Region'),
+                value: currRegion ? currRegion : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: schemasRegions.map(region => ({
+                label: region,
+                alwaysShow: region === currRegion,
+                description:
+                    region === currRegion ? localize('AWS.wizard.selectedPreviously', 'Selected Previously') : ''
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        return val ? (val.label as string) : undefined
+    }
+
+    public async promptUserForRegistry(currRegion: string, currRegistry?: string): Promise<string | undefined> {
+        const client: SchemaClient = ext.toolkitClientBuilder.createSchemaClient(currRegion)
+        const registryNames = await SchemasDataProvider.getInstance().getRegistries(currRegion, client)
```

Comment:
How do you handle client/service errors that may arise from the service call?

Reply:
Good point, I wasn't. Thinking of wrapping the api call inside the SchemasDataProvider class with try catch - if it fails display showErrorMessage and return undefined.  In the wizard, check the if registryNames is undefined, if yes, go back to previous step.  Do the similar flow for schemas api call.